test/models/compute/server_tests.rb

```diff
@@ -0,0 +1,224 @@
+require "test_helper"
+
+describe "Fog::Compute[:openstack] | server" do
+  before do
+    @compute = Fog::Compute[:openstack]
+  end
+
+  describe "success" do
+    it "#floating_ips" do
+      flavor = @compute.flavors.first.id
+      image  = @compute.images.first.id
+      server = @compute.servers.new(
+        :name       => 'test server',
+        :flavor_ref => flavor,
+        :image_ref  => image
+      )
+      server.save
+
+      ip1 = Fog::Network[:openstack].floating_ips.create(
+        :floating_network_id => 'f0000000-0000-0000-0000-000000000000',
+        :fixed_ip_address    => '192.168.11.3'
+      )
+
+      server.associate_address(ip1.fixed_ip_address)
+      server.reload
+      server.floating_ip_addresses.must_equal(["192.168.11.3"])
+    end
+
+    it "#security_groups" do
+      begin
+        my_group = @compute.security_groups.create(
+          :name        => 'my_group',
+          :description => 'my group'
+        )
+
+        flavor = @compute.flavors.first.id
+        image  = @compute.images.first.id
+
+        server = @compute.servers.new(
+          :name       => 'test server',
+          :flavor_ref => flavor,
+          :image_ref  => image
+        )
+
+        server.security_groups = my_group
+
+        server.save
+
+        found_groups = server.security_groups
+        found_groups.length.must_equal 1
+
+        group = found_groups.first
+        group.name.must_equal 'my_group'
+        group.service.must_equal server.service
+      ensure
+        unless Fog.mocking?
+          server.destroy if server
+
+          begin
+            @compute.servers.get(server.id).wait_for { false }
+          rescue Fog::Errors::Error
+            # ignore, server went away
+          end
+        end
+        my_group.destroy if my_group
+      end
```

Comment:
This feels quite strange to have all this code inside of `it` block especially with `begin`-`ensure`-`end`. Also this `it` has more then one assert inside.